### PR TITLE
deps: upgrade to sdk v0.50.x

### PR DESCRIPTION
**Pull request by faddat**
## What is the purpose of the change

* Upgrade Osmosis to v0.50.x
* Identify which of Osmosis' dependencies will need upgrades when Osmosis upgrades to v0.50.x

## Testing and Verifying

For now, there's not a good reason to test this PR signficantly.  It is unlikely to work until an upgraded 
packet-forward-middleware and other dependencies have bene completed. 

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,12 +4,10 @@ go 1.20
 
 require (
 	cosmossdk.io/errors v1.0.0
-	github.com/CosmWasm/wasmd v0.31.0
-	github.com/cosmos/cosmos-proto v1.0.0-beta.2
-	github.com/cosmos/cosmos-sdk v0.47.4
+	github.com/cosmos/cosmos-proto v1.0.0-beta.3
+	github.com/cosmos/cosmos-sdk v0.50.0-beta.0.0.20230802171422-ee5b2e3afc4f
```
Comment (mattverse) [inline go.mod:8]: is this the version with the bank hooks figured out?

**Comment by mattverse** [comment]
Thanks for the PR @faddat, we are still currently figuring out ways to most efficiently unfork sdk whilst having our changes upstreamed: we're also down to have the next major upgrade soley focused on unforking sdk, upgrading to 50 (if possible). But as mentioned, we still have not figured out if we want to upgrade to v0.50, or even if its possible to do so, so my recommendation would be punting further work for this until we have a fixed decision made! wdyt?

**Comment by faddat** [comment]
Hey Matt, actually my thought on this is that it might be best to do an
investigation into it and then provide you guys with the results but also
if you would like us to wait that's totally fine.

I'll bring some thoughts to you guys in slack, I think.

So I think that a unique challenge that we are going to face is that we're
going to have a pretty hard time defining the appropriate steps for the
upgrade without some effort into actually doing the upgrade, but I will get
you some information in slack and wait to hear back from you before
investing a ton more time.

**Comment by mattverse** [comment]
Sounds great, thanks!

**Pull request closed without merging**